api/all/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java

```diff
@@ -69,7 +67,9 @@ private static Baggage sortAndFilterToBaggage(Object[] data) {
 
     @Override
     public BaggageBuilder setParent(Context context) {
-      requireNonNull(context, "context");
+      if (context == null) {
```

Comment:
Why not root?

Reply:
do you mean: why not put the Baggage into the root context? That would seem like a pretty unexpected behavior, since the user didn't do anything with the root context.

Reply:
No I mean using root as the parent, aka no Parent behavior

Reply:
oh, I see. sorry. I was confused. isn't that the same thing as what this is doing, then? You're basically saying that it doesn't have a parent?

Reply:
No, because if I do builder.setParent(nonNullContext).setParent(null) will result in the `nonNullContext` to be the parent.

Reply:
Also I think we should spend a bit more time before doing this change, because I don't think we can easily change this in the future. So we need to make sure this is what we want. Having now throwing NPE allows us to easily change it in the future to no longer throw.

Reply:
To be clear: the null behavior for get is great and I am on board with that, it also matches the `Map.get`

Reply:
Makes sense, reverted it for now (think I forgot that was there by the time I finished the PR). Left the tweaked test case since it'll be needed sooner than latter.